orc/orcarm.c

```diff
@@ -1723,6 +1806,79 @@ orc_arm64_emit_mem (OrcCompiler *p, OrcArm64RegBits bits, OrcArm64Mem opcode,
   orc_arm_emit (p, code);
 }
 
+/** memory access instructions (pair)
+ *
+ *    3                   2                   1
+ *  1 0 9 8 7 6 5 4 3 2 1 0 9 8 7 6 5 4 3 2 1 0 9 8 7 6 5 4 3 2 1 0
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |b 0|1 0 1|0| opt |L|     imm7    |   Rt2   |    Rn   |    Rt   |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ */
+
+#define arm64_code_mem_pair(b,opt,L,imm,Rt2,Rn,Rt) (0x40000000 | ((((b)==64)&0x1)<<31) | \
+    (((opt)&0x7)<<23) | (((L)&0x1)<<22) | (((imm)&0x7f)<<15) | (((Rt2)&0x1f)<<10) | \
+    (((Rn)&0x1f)<<5) | ((Rt)&0x1f))
+
+void
+orc_arm64_emit_mem_pair (OrcCompiler *p, OrcArm64RegBits bits, OrcArm64Mem opcode,
+    int opt, int Rt, int Rt2, int Rn, orc_int32 imm)
+{
+  orc_uint32 code;
+
+  static const char *insn_names[3] = {
+    "stp", "ldp"
```

Comment:
It'd be safer if you use:
[ORC_ARM64_MEM_STR] = "stp",
[ORC_ARM64_MEM_LDR] = "ldp",

Reply:
Yes, I think I forgot to add `opcode -= ORC_ARM64_MEM_STR`.
I wanted to use the index of the opcode is started from zero in this function.

Reply:
You have defined:
```
typedef enum {
  ORC_ARM64_MEM_STR = 0,
  ORC_ARM64_MEM_LDR
} OrcArm64Mem;
```
So, don't worry about that.